bnd-maven-plugin-parent/bnd-maven-plugin/src/main/java/org/bndtools/maven/MavenVersion.java

```diff
@@ -0,0 +1,63 @@
+package org.bndtools.maven;
+
+/**
+ * Represents a Maven version with the ability to translate to a bnd version,
+ * with SNAPSHOT being replaced by bnd's ${tstamp} macro.
+ * 
+ * @author Neil Bartlett <neil.bartlett@paremus.com>
+ */
+public class MavenVersion {
+
+	private static final String SNAPSHOT = "SNAPSHOT";
+	private static final String TSTAMP = "${tstamp}";
+
+	private int major;
+	private int minor;
+	private int micro;
+	private String qualifier;
+	
+	public MavenVersion(String versionString) throws IllegalArgumentException {
```

Comment:
You may want to use the new maven version class: <https://github.com/bndtools/bnd/blob/master/biz.aQute.repository.aether/src/aQute/bnd/deployer/repository/aether/MvnVersion.java>. It handles mapping bazaar maven versions into OSGi versions quite well.  See the tests <https://github.com/bndtools/bnd/blob/master/biz.aQute.repository.aether/test/aQute/bnd/deployer/repository/aether/MvnVersionTest.java>.